src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -435,3 +437,20 @@ def _get_local_storage(self, run: Union[str, Run]) -> LocalStorageOperations:
         if isinstance(run, str):
             run = self.get(name=run)
         return LocalStorageOperations(run)
+
+    def _get_create_or_update_telemetry_values(self, run: Run, **kwargs):
+        flow_obj = load_flow(source=run.flow)
```

Comment:
Is there any way we can avoid doing this dulplicate load here? Maybe move the logic to the run obj and do some cache?

Reply:
good catch, I have already modified it.